### PR TITLE
GH-2996 - add all members to list, except already members

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Fixes issue where only 10 members were being returned for the filtered select list. Also removes those members that are already members of the board.

#### Ticket Link

  Fixes https://github.com/mattermost/focalboard/issues/2996

### CODE REVIEW

File: webapp/src/components/shareBoard/shareBoard.tsx [hunk 1]
```diff
@@ -294,11 +300,12 @@ export default function ShareBoardDialog(props: Props): JSX.Element {
                     <div className='share-input'>
                         <SearchIcon/>
                         <Select
+                            key={JSON.stringify(membersCount)}
```
Comment (sbishel) [inline webapp/src/components/shareBoard/shareBoard.tsx:303]: This forces the Async Select list to reload. I tried wrapping `searchAndFilterTeamUsers` with useCallback() but could not get it to work. I'm open to other suggestions. This is the solution suggested [here](https://github.com/JedWatson/react-select/issues/1581).

Without this, when a user is added. If you click on the drop down list again the user is still listed. Once you perform a search, the list is updated.

Comment (jespino) [reply webapp/src/components/shareBoard/shareBoard.tsx:303]: Why not use the `filterOptions` property of the `Select` component? looks like the right approach here.

Comment (sbishel) [reply webapp/src/components/shareBoard/shareBoard.tsx]: Way better...Thanks.

**Comment by harshilsharma63** [comment]
@sbishel I think the limit of 10 users was intentional and expected. Depending on the team, it can be a huge list. I think this is actually a UX issue in that it's not communicated to the user that you need to search for the items you're looking for. Happened to me too.

@wuwinson why is it required to show all users in that scrolling dropdown? If we make it more explicit that you need to search for things, would that be fine as well? cc @asaadmahmood 

I'm 3/5 on keeping the hardcoded 10 users limit.

**Comment by wuwinson** [comment]
Will discuss with @asaadmahmood @chenilim in our PM/Design meeting this Thursday (May 5th) to figure out some short term and long term solutions to address this. Need to make it clearer to users that they can search for users outside the list of 10 recents displayed. Also consider whether we or not we need to increase the limit above 10.

**Comment by sbishel** [comment]
Closing this PR. I will move comments over the issue.

**Pull request closed without merging**